lib/routes/pixiv/api/get-illusts.ts

```diff
@@ -8,15 +8,30 @@ import queryString from 'query-string';
  * @param {string} token pixiv oauth token
  * @returns {Promise<got.AxiosResponse<{illusts: illust[]}>>}
  */
-export default function getIllusts(user_id, token) {
-    return got('https://app-api.pixiv.net/v1/user/illusts', {
-        headers: {
-            ...maskHeader,
-            Authorization: 'Bearer ' + token,
-        },
-        searchParams: queryString.stringify({
-            user_id,
-            filter: 'for_ios',
-        }),
-    });
+export async function getIllusts(user_id, token, limit) {
+    let offset = 0;
+    const illusts = new Array<any>();
+    let response;
+    let illustsAmount = 0;
+
+    do {
+        response = await got('https://app-api.pixiv.net/v1/user/illusts', {
+            headers: {
+                ...maskHeader,
+                Authorization: 'Bearer ' + token,
+            },
+            searchParams: queryString.stringify({
+                user_id,
+                filter: 'for_ios',
+                offset,
+            }),
+        });
+        illustsAmount = response.data.illusts.length;
+        for (const element of response.data.illusts) {
+            illusts.push(element);
+        }
+        offset += illustsAmount;
+    } while (illustsAmount === 30 && offset < limit);
+
+    return illusts;
```

Comment:
> 很多路由上限10条，无法增大。
> 
> 1. 消耗大量服务器资源。
> 2. 容易被反爬虫。
> 3. 绝大部分过时 feed 没有意义。
>
> _Originally posted by HenryQW in https://github.com/DIYgod/RSSHub/issues/572#issuecomment-416757360_
>
> RSS 是用来获取更新，增大条目无意义
> 
> 特殊需求请在相应模块里自行修改
>
> _Originally posted by DIYgod in https://github.com/DIYgod/RSSHub/issues/572#issuecomment-416802920_
>
> rss 就是用来跟踪内容源的更新内容的，采集全部内容不是rsshub要做的，建议你去找一个爬虫
>
> _Originally posted by yefoenix in https://github.com/DIYgod/RSSHub/issues/7291#issuecomment-809019525_

_Originally posted by TonyRL in https://github.com/DIYgod/RSSHub/discussions/12027#discussioncomment-5184629_

---
> RSS is used to get updates, not history

_Originally posted by DIYgod in https://github.com/DIYgod/RSSHub/issues/3985#issuecomment-586582426_